tests/playground/bool_exhaust_test.lean

```diff
@@ -0,0 +1,731 @@
+import Lean.Elab.Tactic.ElabTerm
+import Lean.Elab.Command
+
+open Lean Lean.Meta Lean.Elab Lean.Elab.Term Lean.Elab.Command
+
+
+
+-- This file runs many tests on simp and other operations on
+-- Boolean/Prop values.
+--
+-- It is intended to systematically evaluate simp strategies on
+-- different operators.
+
+-- Note. These tests use the simp tactic not because simp is the best
+-- strategy for these particular examples, but rather simp may wind up
+-- needing to discharge conditions during rewriting, and we need tests
+-- showing that is has generally effective and predictable behavior.
+
+/-
+General goals for simp are that the normal forms are sensible to a wide
+rnge of users and that it performs well.
+
+Specific goals with Bool are
+1. Consistent behavior with equivalent Bool and Prop operators (e.g, &&
+   and ∧).
+2. Distributivity theorems exist between and, or and not but are not in
+   default simp set.
+3. Negation moves to outside of equality and inequality (xor), but will
+   preserve operator.
```

Comment:
Could you clarify "but will preserve operator"?

Reply:
Good question.  This comment is out of date and reflects this code before I added all the things for Mathlib compatibility.

Reply:
I'm removing it from here for now.